common4j/src/test/com/microsoft/identity/common/java/platform/DeviceTest.java

```diff
@@ -0,0 +1,94 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.java.platform;
+
+import com.microsoft.identity.common.java.AuthenticationConstants;
+import com.microsoft.identity.common.java.logging.DiagnosticContext;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+import lombok.val;
+
+/**
+ * Tests for {@link com.microsoft.identity.common.java.platform.Device}.
+ */
+public class DeviceTest {
+
+    @After
+    public void setUp() {
+        Device.clearDeviceMetadata();
+    }
+
+    @Test
+    public void testGetDataWhenMetadataIsNotSet(){
+        // Shouldn't crash.
+        val platformParameter = Device.getPlatformIdParameters();
+        Assert.assertEquals(3, platformParameter.size());
+        Assert.assertEquals("NOT_SET", platformParameter.get(Device.PlatformIdParameters.CPU_PLATFORM));
```

Comment:
let's define `NOT_SET` in a constant in this file and use that here and everywhere in this test